MIDAS/src/finite-state-machines/fsm.cpp

```diff
@@ -0,0 +1,369 @@
+             	state.curr_state = STATE_MAIN; 
+            }
+            break;
+        
+        default:
+            return state;
+    
+    }
+    return state;
+}
+
+// this is similiar to the previous function but contains less states
+FSMState FSM::tick_fsm_booster(FSMState & state, std::array<HighGData, 8> & hg, std::array<Barometer, 8> & bar){
+
+    double current_time = pdTICKS_TO_MS(xTaskGetTickCount());
+    
+    switch (state.curr_state)
+    {
+        case STATE_IDLE:
+            if(getAcceleration(hg) > idle_to_first_boost_acceleration_threshold) {
+                launch_time = current_time;
+                state.curr_state = STATE_FIRST_BOOST;
+            }
+            
+            break;
+
+        case STATE_FIRST_BOOST:
+            acceleration = getAcceleration(hg);
+            
+            if(acceleration < idle_to_first_boost_acceleration_threshold && current_time - launch_time < idle_to_first_boost_time_threshold) {
+                 state.curr_state = STATE_IDLE;
+                 break;
+            }
+            if(acceleration < coast_detection_acceleration_threshold) {
+                burnout_time = current_time;
+                state.curr_state  = STATE_BURNOUT;
+            }
+
+            break;
+
+        case STATE_BURNOUT:
+            if(getAcceleration(hg) >= coast_detection_acceleration_threshold && (current_time - burnout_time) < first_boost_to_burnout_time_threshold) { 
+                state.curr_state = STATE_FIRST_BOOST;
+                break;
+            }
+
+            if(current_time - burnout_time > first_boost_to_burnout_time_threshold) { 
+                first_seperation_time = current_time;
+                state.curr_state = STATE_FIRST_SEPERATION;
+            }
+            break;
+
+        case STATE_FIRST_SEPERATION: 
+            if (getJerk(hg) < jerk_threshold) {
+                state.curr_state = STATE_COAST;
+                break;
+            }
+
+            if(current_time - first_seperation_time > first_seperation_time_threshold) { 
+	            state.curr_state = STATE_COAST;
+            }
+
+            break;
+            
+        case STATE_COAST:
+            if (getVerticalSpeed(bar) <= 0) {
+             	apogee_time = current_time;
+             	state.curr_state = STATE_APOGEE; 
+            }
+            break;
+        
+        case STATE_APOGEE:
+            if(getVerticalSpeed(bar) > 0 && current_time - apogee_time < apogee_check_threshold) {
+                state.curr_state = STATE_COAST;
+                break;
+            }
+
+            if(current_time - apogee_time > apogee_timer_threshold) {
+                drogue_time = current_time;
+                state.curr_state = STATE_DROGUE_DEPLOY;
+            }
+            break;
+            
+        case STATE_DROGUE_DEPLOY:
+            if(getJerk(hg) < jerk_threshold) { 
```

Comment:
jerk_threshold should be different between the drogue, the main, and the separation. These values are likely different and we can only really verify with testing, but it stands that they should be slightly different.